### PR TITLE
POC - Postgres Push Notifications

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#8838](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8838)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"? 

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

## Description

Implements a Webclient background service that listens for DB changes to Banners in the Communication table.
The active banner is cached according to the expiry date and only then would be re-read from the DB although any insert of an active banner would result in the cache being updated directly without read. 

## Testing

* [X] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce:

N/A

### UI Changes

YES | NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments
Code has been functionally tested on Insert/Update/Delete actions in the DB.

### CODE REVIEW

File: Apps/WebClient/src/WebClient.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <SpaRoot>ClientApp\</SpaRoot>
     <DefaultItemExcludes>$(DefaultItemExcludes);$(SpaRoot)node_modules\**</DefaultItemExcludes>
     <WarningsAsErrors>CS8600;CS8602;CS8603</WarningsAsErrors>
+    <OutputType>Exe</OutputType>
```
Comment (FuriousLlama) [inline Apps/WebClient/src/WebClient.csproj:14]: VS changes?

File: Apps/WebClient/src/Server/Models/BannerChangeEvent.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+//-------------------------------------------------------------------------
+namespace HealthGateway.WebClient.Models
+{
+    using System;
+    using HealthGateway.Database.Models;
+
+    /// <summary>
+    /// Object that represents a banner change push.
+    /// </summary>
+    public class BannerChangeEvent
```
Comment (FuriousLlama) [inline Apps/WebClient/src/Server/Models/BannerChangeEvent.cs:24]: This object looks general enough. Thoughts on making a base change event and subclassing for the data field?

**Comment by FuriousLlama** [comment]
That's a pretty neat pattern! Wonder if we can use for other cache objects like the Keyword

File: Apps/WebClient/src/Server/Models/BannerChangeEvent.cs (hunk 2 above)
Comment (sslaws) [reply Apps/WebClient/src/Server/Models/BannerChangeEvent.cs:24]: Yes, I was working on it as a base class but due to some complexities I removed it for the time being.  When/if we do this again we can then abstract it out.

File: Apps/WebClient/src/WebClient.csproj (hunk 1 above)
Comment (sslaws) [reply Apps/WebClient/src/WebClient.csproj:14]: Yes, frustrating... I'll fix!

File: Apps/WebClient/src/Server/Listeners/BannerListener.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        {
+            this.logger.LogDebug($"Registering Channel Notification on channel {Channel}");
+            stoppingToken.Register(() =>
+                this.logger.LogInformation($"DBChangeListener exiting..."));
+            try
+            {
+                using var scope = this.services.CreateScope();
+                GatewayDbContext dbContext = scope.ServiceProvider.GetRequiredService<GatewayDbContext>();
+                this.CommunicationService = scope.ServiceProvider.GetRequiredService<ICommunicationService>();
+                NpgsqlConnection con = (NpgsqlConnection)dbContext.Database.GetDbConnection();
```
Comment (leobcgov) [inline Apps/WebClient/src/Server/Listeners/BannerListener.cs:83]: Do you think using "using" block is better to protect our db connection from leaking?

**Pull request merged**